internal/generate/clusterserviceversion/clusterserviceversion_test.go

```diff
@@ -359,6 +359,37 @@ var _ = Describe("Generating a ClusterServiceVersion", func() {
 				Expect(csv).To(Equal(upgradeCSV(newCSV, g.OperatorName, g.Version)))
 			})
 		})
+
+		Context("generate ClusterServiceVersion", func() {
```

Comment:
Tests for the functionality added in this PR should be in `internal/generate/collector`, not here.

Reply:
Currently, there are no unit tests exist specific to `internal/generate/collector` functionality. The `clusterserviceversion_test.go` also covers collector functionality, hence I add a new case to test this new change.

Reply:
Well you can add a test to `internal/generate/collector` so specific tests exist :smiley:

Reply:
@estroz Added a test for chagne in `collector` package.

Reply:
@avalluri thanks! Can you also remove this test, since the new functionality is being tested elsewhere?

Reply:
@estroz Thanks for your review time. I intentionally left the test in `clustersevice_version.go` as the test also ensures the `alm-examples` annotations filled properly even for CRs with core type names. I made changes to test such that it makes sure the generated `alm-examples` matches with found CRs.